### DISCUSSION TITLE
[@types/bcrypt] TS80007 error when using hash function with await.

### DISCUSSION THREAD

**Discussion by shahzaib-ai** [DefinitelyTyped/DefinitelyTyped#68644] [Issues with a @types package]
### Problem

`@types/bcrypt`  hash function causing TS80007 error in VS Code (IDE). I think it is happening [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/25a1ebf5c870e8f374283658c17e03f63de73c60/types/bcrypt/index.d.ts#L124)

VS Code (IDE) reports error when using await and if await is not used, a pending promise is returned rather than actual hashed string.

### Screenshot  
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/27653818/3527396f-d9cc-48d7-920d-f87df65392f5)

### Possible fix done on own 

Changing void to Promise<string> fixes the TS80007 error in VS Code (IDE)

From
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/27653818/fa11e5b7-08ac-4ed0-b7e0-7ac850ca9c72)
To
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/27653818/f3d66d34-ffaf-442e-8c6b-4d9e688df9ed)

After Fix 

![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/27653818/97426ffc-f36d-469a-aa5e-77ba4ff8698d)

### Screencast 

[Screencast from 15-02-24 11:15:28 AM PKT.webm](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/27653818/6e67a0c5-05b9-4f5a-a8d5-8546c54dbf4a)

### Code to reproduce

```js
import bcrypt from 'bcrypt';

class DBSchema {
	constructor(password) {
		this.password = password;
	}

	cb(fn) {
		fn.bind(this)();
	}
}

const SchemaInstance = new DBSchema('myplainpass');

SchemaInstance.cb(async function () {
	console.log(this.password);
	this.password = await bcrypt.hash(this.password, 10);
	console.log(this.password);
});

```

[a comment by an automated account is not shown]

**Comment by ghost** [comment]
Fixed in #68646
